repos/system_upgrade/el7toel8/actors/sctpconfigupdate/libraries/sctpupdate.py

```diff
@@ -1,17 +1,25 @@
+import os
+
 from leapp.libraries.stdlib import api, run
 from leapp.models import SCTPConfig
 
 
-def enable_sctp():
+def enable_sctp(_modprobe_d_path="/etc/modprobe.d"):
     """
     Enables the SCTP module by removing it from being black listed.
+    :type _modprobe_d_path: str
+    :param _modprobe_d_path: overwrite only in case of testing, by passing
+        some tmp_dir instead
     """
 
     api.current_logger().info('Enabling SCTP.')
-    run(['/usr/bin/sed', '-i', r's/^\s*blacklist.*sctp/#&/',
-         '/etc/modprobe.d/sctp_diag-blacklist.conf',
-         '/etc/modprobe.d/sctp-blacklist.conf'])
-    api.current_logger().info('Enabled SCTP.')
+    cmd_return = run(['/usr/bin/sed', '-i', r's/^\s*blacklist.*sctp/#&/',
+                      os.path.join(_modprobe_d_path, 'sctp_diag-blacklist.conf'),
+                      os.path.join(_modprobe_d_path, 'sctp-blacklist.conf')])
+    # TODO sed returns 0 even in case the pattern wasn't match. So it is not fair
+    #  to claim that SCTP enabled
+    if cmd_return["exit_code"] == 0:
+        api.current_logger().info('Enabled SCTP.')
```

Comment:
this does not need to be checked, as with non-zero exit code, the `CalledProcessError` will be raised.

Reply:
Resolved